.github/workflows/main.yml

```diff
@@ -0,0 +1,80 @@
+name: pets-app CI/CD
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+
+
+jobs:
+
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: install pre-commit
+        run: pip3 install pre-commit
+      - name: run pre-commit
+        run: pre-commit run --all-files
+
+  build-artifacts:
+    needs: [ pre-commit ]
+    runs-on: ubuntu-latest
+    outputs:
+      file_hash1: ${{ steps.create-package.outputs.file_hash }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: create zip package
+        id: create-package
+        env:
+          file_hash: ${{ hashFiles('app/**', 'requirements.txt') }}
+        run: |
+          pip3 install -r requirements.txt --target dependencies
+          cd ./dependencies && zip -r9 ../$file_hash .
+          # add APP files to Zip file
+          cd .. && zip -g ./$file_hash -r ./app
+          echo "file_hash=$file_hash" >> $GITHUB_OUTPUT
+      - name: Install AWS CLI
```

Comment:
use official aws action instead
https://github.com/marketplace/actions/configure-aws-credentials-action-for-github-actions

you won't need to pass AWS_ACCESS_KEY_ID/AWS_SECRET_ACCESS_KEY/AWS_DEFAULT_REGION in upload to s3 step as the credentials will already be configured